src/interfaces/IAxelarExecutableForetellable.sol

```diff
@@ -0,0 +1,101 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.9;
+
+import { IAxelarGateway } from './IAxelarGateway.sol';
+import { IERC20 } from './IERC20.sol';
+
+abstract contract IAxelarExecutableForetellable {
+    error NotApprovedByGateway();
+    error AlreadyFortold();
+    error NotAuthorizedToForetell();
+
+    IAxelarGateway public gateway;
+    mapping(string => mapping(string => mapping(bytes => mapping(string => mapping(uint256 => address)))))
```

Comment:
```suggestion
    mapping(bytes32 => address) public foretellers;
```
and key becomes keccak256(sourceChain, sourceAddress, payload, tokenSymbol, amount). Fewer hashing happening and therefore cheaper gas cost, no?